tree/tree/src/TTree.cxx

```diff
@@ -4383,53 +4383,56 @@ Int_t TTree::Fill()
 {
    Int_t nbytes = 0;
    Int_t nerror = 0;
-   Int_t nb = fBranches.GetEntriesFast();
-   if (nb == 1) {
-      // Case of one single super branch. Automatically update
-      // all the branch addresses if a new object was created.
-      TBranch* branch = (TBranch*) fBranches.UncheckedAt(0);
-      branch->UpdateAddress();
-   }
-   if (fBranchRef) {
+   Int_t nbranches = fBranches.GetEntriesFast();
```

Comment:
Most changes in the commit "Optimize TTree::Fill()" seems unrelated to the 'modulo operations' making it harder to understand ... is it possible to either split it or at least explain all the changes?